presto-testing/src/main/java/io/prestosql/testing/QueryAssertions.java

```diff
@@ -339,4 +339,39 @@ public static void copyTable(QueryRunner queryRunner, QualifiedObjectName table,
         long rows = (Long) queryRunner.execute(session, sql).getMaterializedRows().get(0).getField(0);
         log.info("Imported %s rows for %s in %s", rows, table.getObjectName(), nanosSince(start).convertToMostSuccinctTimeUnit());
     }
+
+    static RuntimeException getPrestoExceptionCause(Throwable e)
+    {
+        return Throwables.getCausalChain(e).stream()
+                .filter(QueryAssertions::isPrestoException)
+                .findFirst() // TODO .collect(toOptional()) -- should be exactly one in the causal chain
+                .map(RuntimeException.class::cast)
+                .orElseThrow(() -> new IllegalArgumentException("Exception does not have PrestoException cause", e));
+    }
+
+    private static boolean isPrestoException(Throwable exception)
+    {
+        requireNonNull(exception, "exception is null");
+
+        if (exception instanceof PrestoException) {
+            return true;
+        }
+        if (exception instanceof ParsingException) {
+            return true;
```

Comment:
Note: this is needed e.g. because of TestDistributedEngineOnlyQueries / TestLocalEngineOnlyQueries.